### PR TITLE
fix(cirriculum): add better testcase for insert-an-element-into-a-max-heap challenge

### LINKED ISSUE #44269: Faulty code successfully passes tests in Insert an Element Into a Max Heap challenge
**Describe the bug**

 **Insert an Element Into a Max Heap challenge (Coding Interview Prep Section--Data Structures):**
 I erroneously thought my code was working properly because I passed the challenge. But the array resulting from my code does not satisfy the max heap property.
**To Reproduce:**

1. Go to ['Insert an Element Into a Max Heap'](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/insert-an-element-into-a-max-heap)
2. Paste the following code into the space between the //only change code below this line and //only change code above this line comments:
```
      this.heap = [null];
      this.insert = value => {
        if(this.heap.length === 1) {
          this.heap.push(value);
        } else {
          //this.heap.length = the not-yet-existent index after the last item.
        let parentIndex = Math.floor(this.heap.length/2);
        let parentValue = this.heap[parentIndex];
        if(parentValue < value) {
          this.heap[parentIndex] = value;
          this.heap.push(parentValue);
        } else if (parentValue >= value){
          this.heap.push(value);
        }
        
        }
      }
      this.print = () => {
        console.log(this.heap);
        return this.heap;
      }
```
3. Click "Run the Tests" and watch it pass.
4. Test, either on FCC or in some other environment like JSFiddle/CodeSandbox or your own code editor, by copying and pasting the following code after the MaxHeap() function:

```
let mh = new MaxHeap();
  mh.insert(45);
  console.log(mh.heap);
  mh.insert(30);
  console.log(mh.heap);
  mh.insert(35);
  console.log(mh.heap);
  mh.insert(50);
  console.log(mh.heap);
  mh.insert(20);
  console.log(mh.heap);
  mh.insert(15);
  console.log(mh.heap);
```

5. View console. See how the last three printed arrays do not meet the max heap property. For example, [ null, 45, 50, 35, 30, 20, 15 ]. 50 > 45, but 45 is its parent.

**Expected behavior:**
My code should not pass the last test on FCC. The last test states, "The insert method should add elements according to the max heap property."

**Desktop:**
 - OS: Windows 10
 - Browser: Firefox Developer Edition
 - Version: 95.0

### PR DESCRIPTION
**Pull request by Prathamesh010**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #44269

### CODE REVIEW

**Review by gikf — COMMENTED** [review]
This makes code from the issue not pass, but some other faulty solutions can now pass, for example:
```js
var MaxHeap = function() {
  // Only change code below this line
      this.heap = [null];
      this.insert = value => {
        this.heap[1] = value;
      }
      this.print = () => {
        return this.heap;
      }
  // Only change code above this line
};
```

What do you think about checking order of the whole heap (minus possible `null` as first element)? Shouldn't those be the same for any correct implementation?

**Comment by Prathamesh010** [comment]
Yes, Checking the order of the final heap to determine the result is a better way to test it.

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.md [hunk 1]
```diff
@@ -110,8 +110,12 @@ assert(
...
     test.insert(32);
     test.insert(51);
-    let result = test.print();
-    return result.length == 5 ? result[0] == 700 : result[1] == 700;
+    test.insert(800);
+    let result = JSON.stringify(test.print());
+    let solution = JSON.stringify([null,800,51,700,32,50,100]);
+    let solutionWithoutNull = JSON.stringify([800,51,700,32,50,100]);
+
+    return (result.length == 6) ? (result == solution) : (result == solutionWithoutNull);
```
Comment (gikf) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.md:118]: `result` is here stringified version of the heap array, so length `6` will neither be right for version with null nor without.

Assuming length of the heap array will be checked, it should be compared to `7`, as that's how are ordered comparisons with solutions later in the ternary operator.

And from the nitpicking - those variables can be declared with `const`.

**Review by gikf — CHANGES_REQUESTED** [review]
Thanks for the update. I've included couple notes below.

**Comment by Prathamesh010** [comment]
I made the changes. 
Also, I noticed that the provided [solution](https://forum.freecodecamp.org/t/freecodecamp-challenge-guide-insert-an-element-into-a-max-heap/301703) on the forum needs some changes. 

```javascript
while (ele > arr[Math.floor(index / 2)] && index > 1) {
      arr[index] = arr[Math.floor(index / 2)];
      arr[Math.floor(index / 2)] = ele;
      index = arr[Math.floor(index / 2)]; // this should be index = Math.floor(index/2)
}
```

**Review by gikf — APPROVED** [review]
Looks good to me :+1: 

I've tried few implementations, which were giving the expected result as here. If that won't be enough, some alternative could be more manually checking if returned heap array follows max heap conditions.

**Review by naomi-lgbt — APPROVED** [review]
This looks good to me.

**Comment by nhcarrigan** [comment]
Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 🎉

**Pull request merged**